Neovim setup (updated)

baco opened the pull request:
As result of [this PR](https://github.com/neovim/nvim-lspconfig/pull/3078) being accepted and merged; now Neovim configuration is more direct.

There is no need to continue using another LSP server's configuration to be loaded into Neovim via `nvim-lspconfig`.

bluss commented:
I think nvim-lspconfig docs seems to say to use the following:

```
 python = {
    analysis = {
```

but they have been renamed like this, if I'm not mistaken:

```
 basedpyright = {
    analysis = {
```

baco commented:
You are absolutely right.  I've submitted the fixing PR: https://github.com/neovim/nvim-lspconfig/pull/3079

DetachHead commented:
> You are absolutely right. I've submitted the fixing PR: [neovim/nvim-lspconfig#3079](https://github.com/neovim/nvim-lspconfig/pull/3079)

i'm not really familiar with neovim, is that something that'll require another change to these installation instructions once merged, or is this ready to merge now?

baco commented:
On this side is ready now, is just what the documentation says.  In the future, on the adapter plugin (`nvim-lspconfig`) more of the settings available in the LSP server can be added to the schema.

I didn't add them all (on the other project) because I wanted to make the plugin available to use in Neovim right away; and if necessary the config file can be used (`pyrightconfig.json`) to setup the LSP server.

What @bluss was correcting to me was more appropriate on the other PR, the one effectively adding the LSP in the first place, but that's already merged (on `nvim-lspconfig`).  That is why I added a second PR on the other project as well.

This one is ready to merge, because the docs I've modified here don't say anything specific about changing the settings.  They are just for turning the LSP server on.

bluss commented:
Thanks @baco!
